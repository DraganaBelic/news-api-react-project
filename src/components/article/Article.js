import React from "react";
import { useSelector } from "react-redux";
import { Link, withRouter } from "react-router-dom";
import { useTranslation } from "react-i18next";
import {
  ArticleDivStyle,
  ArticleTitleStyle,
  ArticleImageStyle,
  ArticleContentStyle,
} from "./ArticleStyle";

const Article = (props) => {
  const article = useSelector((state) => state.article.currentArticle);
  const { t } = useTranslation();
  const { title, content, urlToImage } = article;
  const { location } = props;
  return (
    <ArticleDivStyle>
      <ArticleTitleStyle>{title}</ArticleTitleStyle>
      <ArticleImageStyle>
        <img alt={title} src={urlToImage}></img>
      </ArticleImageStyle>
      <ArticleContentStyle>{content}</ArticleContentStyle>
      <Link to={`${location.state.prevPath}`}>
        {"<"}
        {t("backToList")}
      </Link>
    </ArticleDivStyle>
  );
};

export default withRouter(Article);
