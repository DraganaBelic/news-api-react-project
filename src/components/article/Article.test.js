import React from "react";
import configureStore from "redux-mock-store";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import Article from "./Article";
import { render } from "@testing-library/react";
import thunk from "redux-thunk";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

const middlewares = [thunk];
const initialState = {
  article: {
    currentArticle: {
      article: {
        title: "Article title",
        content: "Article content",
        urlToImage: "image",
      },
    },
  }
};
export const mockedStore = (state = initialState) => {
  const mockStore = configureStore(middlewares);
  return mockStore(state);
};
jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (s) => s }),
}));
describe("Test Article component", () => {
  const store = mockedStore();
  const history = createMemoryHistory();
  const historyState = {prevPath: "/prevPath"}
  history.push("/", historyState)
  const {container} = render(
    <Provider store={store}>
      <Router history = {history}>
        <Article />
      </Router>
    </Provider>
  );
  it("should render with given state from Redux store", () => {
    expect(container.firstChild).toMatchSnapshot();
  });
});
