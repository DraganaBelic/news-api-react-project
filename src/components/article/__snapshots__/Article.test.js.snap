// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Article component should render with given state from Redux store 1`] = `
<div
  class="ArticleStyle__ArticleDivStyle-s5eji1-0 eTKQyz"
>
  <h2
    class="ArticleStyle__ArticleTitleStyle-s5eji1-1"
  />
  <div
    class="ArticleStyle__ArticleImageStyle-s5eji1-2 jQaNHS"
  >
    <img />
  </div>
  <div
    class="ArticleStyle__ArticleContentStyle-s5eji1-3 ldTrWp"
  />
  <a
    href="/prevPath"
  >
    &lt;
    backToList
  </a>
</div>
`;
